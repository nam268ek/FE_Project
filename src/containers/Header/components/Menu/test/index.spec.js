import React from 'react';
import renderer from 'react-test-renderer';
import Index from '../index';

let source = [];
beforeEach(() => {
  source = [
    {
      id: 11,
      parent_id: null,
      title: 'Edu School',
      slug: 'Edu-School',
      color: '#52c41a',
      description: 'Giáo dục thời 4.0',
      status: '1',
      image: '/uploads/coursecategory/Lĩnh vực1581496679053.png',
      created_at: '2020-02-12 08:37:59',
      updated_at: '2020-03-01 07:36:51',
      childrens: [
        {
          id: 8,
          parent_id: 11,
          title: 'Tiểu Học',
          slug: 'tieu-hoc',
          color: '#ff00fe',
          description: 'Tiểu Học',
          status: '1',
          image: "/uploads/coursecategory/VUCONGTRUONG 's course1575213379852.jpeg",
          created_at: '2019-12-01 22:16:19',
          updated_at: '2020-02-17 09:17:51'
        }
      ]
    }
  ];
});

describe('Header > components > Menu', () => {
  const component = renderer.create(<Index source={source} />);
  const instance = component.getInstance();
  it('should match the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should call showMenu function ', () => {
    instance.showMenu(source);
  });
});
