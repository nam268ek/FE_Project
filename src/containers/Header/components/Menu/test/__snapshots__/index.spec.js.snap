// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header > components > Menu should match the snapshot 1`] = `
<div
  className="mobile-hidden ant-dropdown-link gx-font-weight-medium gx-pointer gx-ml-4 gx-border gx-text-success vz-custom-menu ant-dropdown-trigger"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="gx-ml-3 gx-border-right gx-pr-2 gx-font-weight-semi-bold"
  >
    Danh muÌ£c
  </span>
  <i
    aria-label="icon: down"
    className="anticon anticon-down gx-mx-2"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </i>
</div>
`;
